File: cis_interface/examples/SaM-multi/src/SaM.m
PsiInterface = py.importlib.import_module('cis_interface.interface.PsiInterface');

disp('Hello from Matlab!');
x = onCleanup( @() exit(-1) );

in = PsiInterface.PsiInput('mlinput1');
st = PsiInterface.PsiInput('mlstatic');
out = PsiInterface.PsiOutput('mloutput');
disp('Matlab created channels');

raw_Input = in.recv();
Input = char(raw_Input{2});
disp(['Matlab received ', Input, ' from mlinput1']);
raw_Static = st.recv();

Static = char(raw_Static{2});
disp(['Matlab received ', Static, ' from mlstatic']);
Output = str2num(Input)+str2num(Static);
sOutput = sprintf('%d',Output);
out.send(sOutput);
disp(['Matlab sent ', sOutput, ' to mloutput']);

disp('Goodbye from matlab!');

exit();

File: cis_interface/examples/SaM-multi/src/SaM.py
#!/usr/bin/python
import sys
import signal
from cis_interface.interface.PsiInterface import PsiInput, PsiOutput


def signal_handler(signal, frame):
    print('You pressed Ctrl+C!')
    sys.exit(0)

    
if __name__ == '__main__':
    signal.signal(signal.SIGINT, signal_handler)

    print('Hello from Python!')
    in1 = PsiInput('pyinput1')
    in2 = PsiInput('pystatic')
    out1 = PsiOutput('pyoutput')

    adata = in1.recv()[1]
    print("Python received {} from pyinput1".format(adata))
    bdata = in2.recv()[1]
    print("Python received {} from pystatic".format(bdata))

    a = int(adata)
    b = int(bdata)
    sum = a + b

    outdata = str(sum)
    out1.send(outdata)
    print("Python sent {} to pyoutput".format(outdata))

    print('Goodbye from Python!')
